crates/bevy_gltf/src/loader.rs

```diff
@@ -2079,29 +2262,86 @@ mod test {
 
     #[test]
     fn node_hierarchy_cyclic() {
-        let result = resolve_node_hierarchy(
-            vec![
-                (GltfNode::with_generated_name(1), vec![1]),
-                (GltfNode::with_generated_name(2), vec![0]),
-            ],
-            PathBuf::new().as_path(),
-        );
-
-        assert_eq!(result.len(), 0);
+        let gltf_path = "test.gltf";
+        let gltf_str = r#"
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "nodes": [
+        {
+            "name": "l1",
+            "children": [1]
+        },
+        {
+            "name": "l2",
+            "children": [0]
+        }
+    ],
+    "scene": 0,
+    "scenes": [{ "nodes": [0] }]
+}
+"#;
+
+        let dir = Dir::default();
+        dir.insert_asset_text(Path::new(gltf_path), gltf_str);
+        let mut app = test_app(dir);
+        app.update();
+        let asset_server = app.world().resource::<AssetServer>().clone();
+        let handle: Handle<Gltf> = asset_server.load(gltf_path);
+        let handle_id = handle.id();
+        app.world_mut().spawn(handle.clone());
+        app.update();
+        run_app_until(&mut app, |_world| {
+            let load_state = asset_server.get_load_state(handle_id).unwrap();
+            if matches!(load_state, LoadState::Failed(_)) {
+                Some(())
+            } else {
+                None
+            }
+        });
+        assert!(true);
     }
 
     #[test]
     fn node_hierarchy_missing_node() {
```

Comment:
This test is technically testing gLTF library now but I think it's good to keep because we rely on this behaviour to stay.